common/include/dummy_protocol.h

```diff
@@ -22,13 +22,12 @@ class DummyProtocol: public AbstractProtocol {
 
 protected:
 	virtual bool IsFrameComplete(const vector<uint8_t>& buffer) {return true;};
-	virtual bool IsFrameValid(const vector<uint8_t>& buffer) {return true;};
+	virtual bool ExtractValidData(const vector<uint8_t>& buffer,\
+					vector<uint8_t>& unpacked_data)  {return true;};
```

Comment:
Please pay attention to the white space, should be:
```suggestion
					vector<uint8_t>& unpacked_data) {return true;};
```